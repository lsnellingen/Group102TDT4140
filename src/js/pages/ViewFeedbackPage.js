import React from 'react';
import DocumentTitle from 'react-document-title';
import { UserProfileForm } from 'react-stormpath';

export default class ViewFeedbackPage extends React.Component {
  render() {
  	return (
  		<DocumentTitle title={`View Feedback`}>
  			<div className="container">
	        	<div className="row">
	        		<div className="col-xs-12">
		              <h3>View Feedback</h3>
		              <hr />
	        		</div>
	        	</div>
	        	<div className="form-group">
		        	<div className="row">
			            <div className="col-xs-12">
			            	<textarea type="feedback" className="form-control" id="feedback" name="feedback" placeholder="Choose subject and lecture to view feecback... " required />
			            </div>
		            </div>
	            </div>
	            <div className="form-group">
	            	<div className="row">
		            	<div className="col-xs-12 col-sm-12">
				            <p>Choose the subject to view feedback: </p>
				            <select type="subject-selector" className="form-control" id="subject-selector" name="subject-selector">
							  <option value="chooseSubject">Choose subject</option>
							  <option value="subject1">Subject 1</option>
							  <option value="subject2">Subject 2</option>
							  <option value="subject3">Subject 3</option>
							  <option value="subject4">Subject 4</option>
							</select>
						</div>
					</div>
				</div>
				<div className="form-group">
	            	<div className="row">
		            	<div className="col-xs-12 col-sm-12">
				            <p>Choose lecture in chosen subject: </p>
				            <select type="lecture-selector" className="form-control" id="lecture-selector" name="lecture-selector">
				              <option value="chooseLecture">Choose lecture</option>
							  <option value="subject1">Lecture 1</option>
							  <option value="subject2">Lecture 2</option>
							  <option value="subject3">Lecture 3</option>
							  <option value="subject4">Lecture 4</option>
							</select>
						</div>
					</div>
				</div>

	            <div className="form-group">
		            	<button type="button" className="btn btn-primary">View feedback</button>
	            </div>
        	</div>
  		</DocumentTitle>
  	);
  }
}