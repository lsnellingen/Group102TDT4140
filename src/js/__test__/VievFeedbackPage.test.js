import React from 'react';
import ReactDOM from 'react-dom';
import ViewFeedbackPage from '../pages/ViewFeedbackPage';
import renderer from 'react-test-renderer';
import { RegistrationForm, LoginLink } from 'react-stormpath';
import {shallow} from 'enzyme';

describe('ViewFeedbackPage', () => {
  var ReactTestUtils = require('react-addons-test-utils') // ES5 with npm

    test('renders without crashing', () => {
      /*const renderer = ReactTestUtils.createRenderer();*/
      const context={user:{email:'hans@ntnu.no',customData:{courses:"TDT4120 - Algoritmer og datastrukturer"}}};
      const tree = shallow(
        <ViewFeedbackPage/>, {context});


      expect(tree).toMatchSnapshot();




    });
    test('render with one feedback', () => {
      /*user:{email:"larsnee@gmail.com"}*/
      const context={user:{email:'hans@ntnu.no',customData:{courses:"TDT4120 - Algoritmer og datastrukturer"}}};
      var view = ReactTestUtils.renderIntoDocument(<ViewFeedbackPage/>, {context});

    });
});
