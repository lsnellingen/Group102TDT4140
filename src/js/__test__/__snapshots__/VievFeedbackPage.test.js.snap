// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewFeedbackPage renders without crashing 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <SideEffect(DocumentTitle)
    title="View Feedback"
>
    <div
        className="container"
    >
        <div
            className="row"
        >
            <div
                className="col-xs-12"
            >
                <br />
                <h3>
                    View Feedback
                </h3>
                <hr />
            </div>
        </div>
        <div
            className="col-xs-12 col-sm-12"
        >
            <p>
                Choose the course to see feedback in: 
            </p>
            <select
                className="form-control"
                id="subject-selector"
                name="subject-selector"
                onChange={[Function]}
                type="subject-selector"
                value="chooseCourse"
            >
                <option
                    value="chooseCourse"
                >
                    Choose course
                </option>
            </select>
            <hr />
        </div>
        <div
            className="col-sm-10 col-xs-12"
        >
            <div>
                <h5>
                    Choose an option to view feedback.
                </h5>
            </div>
        </div>
    </div>
</SideEffect(DocumentTitle)>,
  "nodes": Array [
    <SideEffect(DocumentTitle)
      title="View Feedback"
>
      <div
            className="container"
      >
            <div
                  className="row"
            >
                  <div
                        className="col-xs-12"
                  >
                        <br />
                        <h3>
                              View Feedback
                        </h3>
                        <hr />
                  </div>
            </div>
            <div
                  className="col-xs-12 col-sm-12"
            >
                  <p>
                        Choose the course to see feedback in: 
                  </p>
                  <select
                        className="form-control"
                        id="subject-selector"
                        name="subject-selector"
                        onChange={[Function]}
                        type="subject-selector"
                        value="chooseCourse"
                  >
                        <option
                              value="chooseCourse"
                        >
                              Choose course
                        </option>
                  </select>
                  <hr />
            </div>
            <div
                  className="col-sm-10 col-xs-12"
            >
                  <div>
                        <h5>
                              Choose an option to view feedback.
                        </h5>
                  </div>
            </div>
      </div>
</SideEffect(DocumentTitle)>,
  ],
  "options": Object {
    "context": Object {
      "user": Object {
        "customData": Object {
          "courses": "TDT4120 - Algoritmer og datastrukturer",
        },
        "email": "hans@ntnu.no",
      },
    },
  },
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {
        "user": Object {
          "customData": Object {
            "courses": "TDT4120 - Algoritmer og datastrukturer",
          },
          "email": "hans@ntnu.no",
        },
      },
      "_currentElement": <ViewFeedbackPage />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": ViewFeedbackPage {
        "_reactInternalInstance": [Circular],
        "context": Object {
          "user": Object {
            "customData": Object {
              "courses": "TDT4120 - Algoritmer og datastrukturer",
            },
            "email": "hans@ntnu.no",
          },
        },
        "handleFiltering": [Function],
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "feedback": Array [],
          "filterOption": "chooseCourse",
          "filteredFeedback": Array [],
          "myCourses": Array [],
          "showWarning": false,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <SideEffect(DocumentTitle)
          title="View Feedback"
>
          <div
                    className="container"
          >
                    <div
                              className="row"
                    >
                              <div
                                        className="col-xs-12"
                              >
                                        <br />
                                        <h3>
                                                  View Feedback
                                        </h3>
                                        <hr />
                              </div>
                    </div>
                    <div
                              className="col-xs-12 col-sm-12"
                    >
                              <p>
                                        Choose the course to see feedback in: 
                              </p>
                              <select
                                        className="form-control"
                                        id="subject-selector"
                                        name="subject-selector"
                                        onChange={[Function]}
                                        type="subject-selector"
                                        value="chooseCourse"
                              >
                                        <option
                                                  value="chooseCourse"
                                        >
                                                  Choose course
                                        </option>
                              </select>
                              <hr />
                    </div>
                    <div
                              className="col-sm-10 col-xs-12"
                    >
                              <div>
                                        <h5>
                                                  Choose an option to view feedback.
                                        </h5>
                              </div>
                    </div>
          </div>
</SideEffect(DocumentTitle)>,
        "_debugID": 2,
        "_renderedOutput": <SideEffect(DocumentTitle)
          title="View Feedback"
>
          <div
                    className="container"
          >
                    <div
                              className="row"
                    >
                              <div
                                        className="col-xs-12"
                              >
                                        <br />
                                        <h3>
                                                  View Feedback
                                        </h3>
                                        <hr />
                              </div>
                    </div>
                    <div
                              className="col-xs-12 col-sm-12"
                    >
                              <p>
                                        Choose the course to see feedback in: 
                              </p>
                              <select
                                        className="form-control"
                                        id="subject-selector"
                                        name="subject-selector"
                                        onChange={[Function]}
                                        type="subject-selector"
                                        value="chooseCourse"
                              >
                                        <option
                                                  value="chooseCourse"
                                        >
                                                  Choose course
                                        </option>
                              </select>
                              <hr />
                    </div>
                    <div
                              className="col-sm-10 col-xs-12"
                    >
                              <div>
                                        <h5>
                                                  Choose an option to view feedback.
                                        </h5>
                              </div>
                    </div>
          </div>
</SideEffect(DocumentTitle)>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ViewFeedbackPage />,
}
`;
